### ISSUE TITLE
Incorrect prop type in shouldForwardProp in emotion 11

### ISSUE THREAD

**Issue by StevenVerbiest** [emotion-js/emotion#2220]
When trying to prevent multiple props from being forwarded, I'm getting a typescript error.

I'm trying to exclude these props by using the following pattern:
```
shouldForwardProp: (prop) => !["prop1", "prop2"].includes(prop)
```

But I'm getting the following error
```
Argument of type 'string | number | symbol' is not assignable to parameter of type 'string'.
  Type 'number' is not assignable to type 'string'.ts(2345)
```
I've managed to reproduce this in a codesandbox:
https://codesandbox.io/s/stupefied-ardinghelli-30pz6?file=/src/StyledComponent.tsx:164-236

This used to work in version 10, where prop was typed as a `string`. 
You can see it working here:
https://codesandbox.io/s/eloquent-dhawan-bwe2x?file=/src/StyledComponent.tsx:259-300

**Expected behavior:**
This should work. 

**Environment information:**

```
{
  "react": "17.0.0", // also occurs in with React 16.13.x
  "@emotion/react": "11.1.x",
  "@emotion/styled": "^11.0.0"
}
```

**Comment by Andarist** [comment]
I think we could relax the prop to be of type `string` instead of `PropertyKey`. We can track this back to a discussion [here](https://github.com/emotion-js/emotion/issues/1641) and the PR [here](https://github.com/emotion-js/emotion/pull/1643) - I don't think there is any strong argument there to actually prefer `PropertyKey`. Thoughts @JakeGinnivan ?

**Comment by StevenVerbiest** [comment]
Ah, I didn't find the issue you linked when trying to find a solution for this. 
I'm indeed working around this by adding an explicit `string` type to the prop as [suggested](https://github.com/emotion-js/emotion/issues/1641#issuecomment-554934623).
This works, but it seems like an unnecessary step. I'm confused how a prop could be a `number` or a `symbol`? Because we are comparing the prop name here and not the prop value if I'm not mistaken.

**Comment by Andarist** [comment]
> I'm confused how a prop could be a number or a symbol?

Yep, this shouldn't happen - we should be able to change that `PropertyKey` to `string`. Are you willing to prepare a PR for this?

**Comment by julix-unity** [comment]
Is this still a valid issue? I see `PropertyKey` mentioned only in the [changelog](https://github.com/emotion-js/emotion/blob/f3b268f7c52103979402da919c9c0dd3f9e0e189/packages/is-prop-valid/CHANGELOG.md?plain=1#L23C8-L23C8) - I encountered this recently, but I guess I should check what version we're on. 

Since it was in the context of a MUI component we just passed `sx` for now. Might revisit this later when not in headsdown mode... 

![Screenshot 2023-11-13 at 2 43 39 PM](https://github.com/emotion-js/emotion/assets/69219981/b7378ba1-b26a-4f04-8427-694c445434c9)
